CRM/Utils/Type.php

```diff
@@ -421,8 +421,8 @@ public static function validate($data, $type, $abort = TRUE, $name = 'One of par
       case 'Date':
       case 'Timestamp':
         // a null timestamp is valid
-        if (strlen(trim($data)) == 0) {
-          return trim($data);
+        if (strlen(trim($data ?? '')) == 0) {
+          return trim($data ?? '');
```

Comment:
I'm just wondering about this one - perhaps the functions calling this care about the difference between NULL & '' - maybe an early return is safer here?

Reply:
in php 7.4 this is the output of the current code

```
seamus@civicrmdevelopment:~/buildkit/build/47-test/web/sites/all/modules/civicrm$ cv ev '$test=trim(NULL);var_dump($test);'
string(0) ""
```